### PR TITLE
Avoid panicing when substracting Instants

### LINKED ISSUE #1209: Bug 1727148
Bug [1727148](https://bugzilla.mozilla.org/show_bug.cgi?id=1727148)

This may be:
https://searchfox.org/mozilla-central/rev/4cca5d2f257c6f1bcef50a0debcbd66524add703/third_party/rust/neqo-transport/src/recovery.rs#651

### PR DESCRIPTION
**Pull request by ddragana**
Fixes #1209

### CODE REVIEW

**Comment by ddragana** [comment]
I also did an audit of the code and I could not fined any other case of substracting time.

File: neqo-transport/src/recovery.rs [hunk 1]
```diff
@@ -640,8 +640,9 @@ impl LossRecovery {
         ack_delay: Duration,
     ) {
         let confirmed = self.confirmed_time.map_or(false, |t| t < send_time);
-        let sample = now - send_time;
-        rtt.update(&mut self.qlog, sample, ack_delay, confirmed, now);
+        if now >= send_time {
+            rtt.update(&mut self.qlog, now - send_time, ack_delay, confirmed, now);
```
Comment (martinthomson) [inline neqo-transport/src/recovery.rs:644]: I think that this is more idiomatic.

```suggestion
        if Some(sample) = now.checked_sub(send_time) {
            rtt.update(&mut self.qlog, sample, ack_delay, confirmed, now);
```

**Review by martinthomson — COMMENTED** [review]
LGTM

File: neqo-transport/src/recovery.rs (hunk 1 above)
Comment (ddragana) [reply neqo-transport/src/recovery.rs:644]: I have tried that, but it does not work checked_sub needs Duration; the output of checked_sub is Instant, but update() needs Duration.

Comment (martinthomson) [reply neqo-transport/src/recovery.rs:644]: My mistake, I should have checked types.  What I meant was:

```suggestion
        if Some(sample) = now.checked_duration_since(send_time) {
            rtt.update(&mut self.qlog, sample, ack_delay, confirmed, now);
```

That is in rust 1.39, so it available in our minimum version.

**Pull request merged**